docs/Guide.TypeScriptIntegrationJest.md

```diff
@@ -0,0 +1,111 @@
+---
+id: Guide.TypeScriptJest
+title: Integrating TypeScript with Detox + Jest
+---
+
+## Usage
+
+### 0. Initialize Detox in your project
+
+Follow the [Getting Started](Introduction.GettingStarted.md) Guide to set up Detox with [Jest as your test runner](Guide.Jest.md) (optionally, set up Detox to [test Android](Introduction.Android.md) as well). When you are finished with that, we will make some changes to your project:
+
+- Remove Detox from `package.json` at the root level
+
+- Create a `package.json` in the `e2e` folder and add Detox and Jest to the project dependencies:
+
+```sh
+npm uninstall detox
+cd e2e
+npm init -y
+npm install -D detox jest
+```
+
+You should then move the settings for Detox from the root-level `package.json` to the `e2e/package.json`, but make sure that you update the `build` and `binaryPath` values for your Detox configurations:
+
+```json
+{
+  // ...
+  "detox": {
+    "configurations": {
+      "ios.sim.debug": {
+        "binaryPath": "../ios/build/Build/Products/Debug-iphonesimulator/YourAppName.app",
+        "build": "xcodebuild -project ../ios/YourAppName.xcodeproj -scheme YourAppName -configuration Debug -sdk iphonesimulator -derivedDataPath ../ios/build",
+        "type": "ios.simulator",
+        "name": "iPhone X"
+      },
+      "android.emu.debug": {
+        "binaryPath": "../android/app/build/outputs/apk/debug/app-debug.apk",
+        "build": "cd ../android; ./gradlew assembleDebug assembleAndroidTest -DtestBuildType=debug; cd ../e2e",
+        "type": "android.emulator",
+        "name": "Pixel_2_API_27"
+      },
+      "android.emu.release": {
+        "binaryPath": "../android/app/build/outputs/apk/release/app-release.apk",
+        "build": "cd ../android; ./gradlew assembleRelease assembleAndroidTest -DtestBuildType=release; cd ../e2e",
+        "type": "android.emulator",
+        "name": "Pixel_2_API_27"
+      }
+    }
+  }
+}
+```
+
+Before moving on, make sure that you can still successfully run your Detox tests.
+
+### 1. Add TypeScript + `ts-jest` to `e2e/package.json`
+
+```sh
+npm install --save-dev typescript ts-jest
+```
+
+### 2. Configure Jest to use `ts-jest`
+
+Modify your Jest configuration (`e2e/config.json` by default) to include the following properties
+
+```json
+{
+  "preset": "ts-jest",
+  "testEnvironment": "node",
+  "setupTestFrameworkScriptFile": "./init.ts"
+}
+```
+
+NB: this is mostly the same output of running `ts-jest config:init`, with `setupTestFrameworkScriptFile` being the only added property.
+
+### 3. `.js` -> `.ts`
+
+Convert all files in the `e2e` directory ending in `.js` to `.ts`
+
+### 4. Add typings for Detox, Jest, and Jasmine
```

Comment:
Please mention @types/detox is maintained by the community, not the owners of this repo.

Reply:
Added a note below